controller/config_test.go

```diff
@@ -311,3 +314,48 @@ func (s *ConfigSuite) TestAuditLogExcludeMethodsType(c *gc.C) {
 	)
 	c.Assert(err, gc.ErrorMatches, `audit-log-exclude-methods\[0\]: expected string, got int\(2\)`)
 }
+
+func (s *ConfigSuite) TestConfigManagementSpaceAsConstraint(c *gc.C) {
+	managementSpace := "management-space"
+	cfg, err := controller.NewConfig(
+		testing.ControllerTag.Id(),
+		testing.CACert,
+		map[string]interface{}{controller.JujuHASpace: managementSpace},
+	)
+	c.Assert(err, jc.ErrorIsNil)
+	c.Assert(*(cfg.AsSpaceConstraints(nil).Spaces), gc.DeepEquals, []string{managementSpace})
```

Comment:
all these get a lot nicer if we just work on a []string
you *might* want to use jc.DeepEquals vs gc.DeepEquals. IIRC the only way in which they differ is that
([]string(nil), gc.DeepEquals, []string{})
fails
while
([]string(nil), jc.DeepEquals, []string{})
says true.

Reply:
As another small thing that I'd like to encourage in the code base, when you can use
c.Check
vs
c.Assert.
The idea is that if following activities still make sense even if this fails, use Check. if they don't, then use Assert.
So often "c.Assert(err, jc.ErrorIsNil)" is used, because if err is not nil, then you likely have a nil pointer, etc, so we want to stop the test there.
But when you can, it is nice to get all the things wrong with your test at once, rather than
run the test, 1 assert fails,
fix, run the test, next assert fails
fix, run the test, next assert fails

with Check, you can see all the bad values in the first run.

Reply:
Acknowledged. Replaced Assert with Check where suitable.